src/modules/user_grants.json

```diff
@@ -0,0 +1,534 @@
+                                "lifespan": "powerActive"
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+    ],
+    "components": {
+        "schemas": {
+            "GrantInfo": {
+                "description": "Information about a grant given by a user",
+                "type": "object",
+                "properties": {
+                    "app": {
+                        "$ref": "#/components/schemas/AppInfo"
+                    },
+                    "state": {
+                        "$ref": "#/components/schemas/GrantState"
+                    },
+                    "capability": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Capability"
+                    },
+                    "role": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Role"
+                    },
+                    "lifespan": {
+                        "type": "string",
+                        "enum": [
+                            "once",
+                            "forever",
+                            "appActive",
+                            "powerActive",
+                            "seconds"
+                        ]
+                    },
+                    "expires": {
+                        "type": "string",
+                        "format": "date-time"
+                    }
+                },
+                "additionalProperties": false,
+                "required": [
+                    "state",
+                    "capability",
+                    "role",
+                    "lifespan"
+                ],
+                "examples": [
+                    {
+                        "app": {
+                            "id": "certapp",
+                            "title": "Firebolt Certification"
+                        },
+                        "state": "active",
+                        "capability": "xrn:firebolt:capability:data:app-usage",
+                        "role": "use",
+                        "lifespan": "seconds",
+                        "expires": "2022-12-14T20:20:39+00:00"
+                    }
+                ]
+            },
+            "AppInfo": {
+                "description": "Information about an app that a grant was for",
+                "type": "object",
+                "properties": {
+                    "id": {
+                        "type": "string"
+                    },
+                    "title": {
+                        "type": "string"
+                    }
+                },
+                "additionalProperties": false,
+                "required": [
+                    "id"
+                ]
+            },
+            "GrantState": {
+                "description": "The state the grant is in",
+                "type": "string",
+                "enum": [
+                    "active",
```

Comment:
active seems more implicit in terms of saying its granted + alive . But then we are trying to push lifespan into a state wouldn't that mean we need to carry state like expired?

Reply:
Changed active to "granted"